### DISCUSSION TITLE
ISO8583 payments MAC Calculator (ISO9797 Alg 3)

### DISCUSSION THREAD

**Discussion by georgelanes** [krzyzanowskim/CryptoSwift#938] [General]
Hi

How can I use this framework to calculate Retail MAC (Message Authentication Code) or ISO9797 Alg 3 ?

Example:
Mac Key: 0123456789abcdefFEDCBA9876543210
Data: 4E6F77206973207468652074696D6520666F7220616C6C20
Alg: DES
Final: 3DES
Result: BDF12AF42D80F70C

Thanks,

George

**Comment by Craz1k0ek** [comment]
As far as I can tell DES/3DES is not supported in this library

**Comment by georgelanes** [comment]
> As far as I can tell DES/3DES is not supported in this library

You know some library write with swift that have support to DES/3DES?
